aspnetcore/release-notes/aspnetcore-5.0.md

```diff
@@ -0,0 +1,495 @@
+
+### Authorization when using endpoint routing
+
+Authorization when using endpoint routing now receives the `HttpContext` rather than the endpoint instance. This allows the authorization middleware to access the `RouteData` and other properties of the `HttpContext` that were not accessible though the <xref:Microsoft.AspNetCore.Http.Endpoint> class. The endpoint can be fetched from the context using [context.GetEndpoint(xref:Microsoft.AspNetCore.Http.EndpointHttpContextExtensions.GetEndpoint%2A).
+
+## API and ASP.NET Core MVC improvements
+
+### Model binding improvements
+
+#### Model binding DateTime as UTC
+
+Model binding now supports binding UTC time strings to `DateTime`. If the request contains a UTC time string, model binding binds it to a UTC `DateTime`. For example, the following time string is bound the UTC `DateTime`:  `https://example.com/mycontroller/myaction?time=2019-06-14T02%3A30%3A04.0576719Z`
+
+#### Model binding and validation with C# 9 record types
+
+[C# 9 record types](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#record-types) can be used with model binding in an MVC controller or a Razor Page. Record types are a great way to model data being transmitted over the network.
+
+For example, the following `PersonController` uses the `Person` record type with model binding and form validation:
+
+```csharp
+public record Person([Required] string Name, [Range(0, 150)] int Age);
+
+public class PersonController
+{
+   public IActionResult Index() => View();
+
+   [HttpPost]
+   public IActionResult Index(Person person)
+   {
+          // ...
+   }
+}
+```
+
+The `Person/Index.cshtml` file:
+
+```cshtml
+@model Person
+
+Name: <input asp-for="Model.Name" />
+<span asp-validation-for="Model.Name" />
+
+Age: <input asp-for="Model.Age" />
+<span asp-validation-for="Model.Age" />
+```
+
+### Control Startup class activation
+
+An additional <xref:Microsoft.AspNetCore.Hosting.HostingAbstractionsWebHostBuilderExtensions.UseStartup%2A> overload has been added that lets an app provide a factory method for controlling `Startup` class activation. Controlling `Startup` class activation is useful to pass additional parameters to `Startup` that are initialized along with the host:
+
+```csharp
+public class Program
+{
+    public static async Task Main(string[] args)
+    {
+        var logger = CreateLogger();
```

Comment:
@Tratcher  who can point me to a test sample that compiles using the new `Startup` overload?

Reply:
https://github.com/dotnet/aspnetcore/commit/113805aba8162ba7bac42943bcb1324801c20f46#diff-23a36744be1739963b7889a34c271f4406b24689afd6cf6329600d7894d01c64R131

Reply:
@Tratcher  that's great for a test, but now how a customer would call it.

Reply:
What's wrong with the sample you already have?

Reply:
> What's wrong with the sample you already have?

I'll use that, but it's not complete. I like to have a complete sample with delegates defined.